fitbenchmarking/parsing/horace_parser.py

```diff
@@ -0,0 +1,194 @@
+            p_names = [k for k in pfg if k != 'foreground']
+            equations = "foreground: " + \
+                os.path.splitext(os.path.basename(pfg['foreground']))[0] +\
+                "<br>" + 'parameters: ' + ', '.join(p_names)
+            start_values = [{n: pfg[n] for n in p_names}]
+        else:
+            pfg = self._parsed_func[0]
+            pbg = self._parsed_func[1]
+            p_names = [k for k in pfg if k != 'foreground']
+            bp_names = [k for k in pbg if k != 'background']
+            equations = "foreground: " + \
+                os.path.splitext(os.path.basename(pfg['foreground']))[0] +\
+                "<br>" + 'parameters: ' + ', '.join(p_names) +\
+                "<br>" + "background: " +\
+                os.path.splitext(os.path.basename(pbg['background']))[0] \
+                + '<br>' + 'parameters: ' + ', '.join(bp_names)
+            fg_starting_vals = {n: pfg[n] for n in p_names}
+            bg_starting_vals = {n: pbg[n] for n in bp_names}
+            start_values = [{**fg_starting_vals, **bg_starting_vals}]
+
+        self._equation = equations
+        self._starting_values = start_values
+
+        simulate_f = self._parse_function(self._entries['simulate_function'])
+        path = os.path.join(os.path.dirname(self._filename),
+                            simulate_f[0]['matlab_script'])
+        eng.addpath(os.path.dirname(path))
+        simulate_func_name = os.path.basename(path).split('.', 1)[0]
+
+        def fit_function(x, *p):
+            # Assume, for efficiency, matching shape => matching values
+            # print(*p)
+            if x.shape != self._horace_x.shape:
+                return np.ones(x.shape)
+            eng.evalc(f'global {self._horace_msk}')
+            eng.evalc(f'global {self._horace_w}')
+            eng.workspace['fitpars'] = matlab.double(p)
+            eng.evalc(f'horace_y = {simulate_func_name}'
+                      f'({self._horace_w},fitpars,{self._horace_msk})')
+            return np.array(eng.workspace['horace_y'],
+                            dtype=np.float64).flatten()
+
+        return fit_function
+
+    def _get_equation(self) -> str:
+        """
+        Returns the equation in the problem definition file.
+
+        :return: The equation in the problem definition file.
+        :rtype: str
+        """
+        return self._equation
+
+    def _get_starting_values(self) -> list:
+        """
+        Returns the starting values for the problem.
+
+        :return: The starting values for the problem.
+        :rtype: list
+        """
+        return self._starting_values
+
+    def _set_additional_info(self) -> None:
+        """
+        Add set_tbf to the fitting problem.
+        """
+        def set_persistent_vars(path):
+            """
+            Update the persistent_vars from a matlab dump.
+
+            :param path: The file to update from
+            :type path: str
+            """
+            horace_on()
```

Comment:
Is this meant to be indented like this?

Reply:
Yes, it's defining a dynamic function which is added to the fitting problem.